src/test/java/com/github/mtakaki/credentialstorage/hibernate/AbstractDAOTest.java

```diff
@@ -0,0 +1,204 @@
+package com.github.mtakaki.credentialstorage.hibernate;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.Serializable;
+import java.util.List;
+
+import org.hibernate.Criteria;
+import org.hibernate.HibernateException;
+import org.hibernate.Query;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.proxy.HibernateProxy;
+import org.hibernate.proxy.LazyInitializer;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.google.common.collect.ImmutableList;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AbstractDAOTest {
+    private static class MockDAO extends BundleAbstractDAO<String> {
+        public MockDAO(final RemoteCredentialHibernateBundle<?> bundle) {
+            super(bundle);
+        }
+
+        @Override
+        public Session currentSession() {
+            return super.currentSession();
+        }
+
+        @Override
+        public Criteria criteria() {
+            return super.criteria();
+        }
+
+        @Override
+        public Query namedQuery(final String queryName) throws HibernateException {
+            return super.namedQuery(queryName);
+        }
+
+        @Override
+        public Class<String> getEntityClass() {
+            return super.getEntityClass();
+        }
+
+        @Override
+        public String uniqueResult(final Criteria criteria) throws HibernateException {
+            return super.uniqueResult(criteria);
+        }
+
+        @Override
+        public String uniqueResult(final Query query) throws HibernateException {
+            return super.uniqueResult(query);
+        }
+
+        @Override
+        public List<String> list(final Criteria criteria) throws HibernateException {
+            return super.list(criteria);
+        }
+
+        @Override
+        public List<String> list(final Query query) throws HibernateException {
+            return super.list(query);
+        }
+
+        @Override
+        public String get(final Serializable id) {
+            return super.get(id);
+        }
+
+        @Override
+        public String persist(final String entity) throws HibernateException {
+            return super.persist(entity);
+        }
+
+        @Override
+        public <T> T initialize(final T proxy) {
+            return super.initialize(proxy);
+        }
+    }
+
+    @Mock
+    private RemoteCredentialHibernateBundle<?> bundle;
+    @Mock
+    private SessionFactory factory;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/mitsuotakaki/CredentialStorageService-dw-hibernate/file/2455857653/issues/source?bid=3237716&fileBranchId=3250434#l93)